client/state/sites/selectors.js

```diff
@@ -517,3 +518,95 @@ export function isCurrentSitePlan( state, siteId, planProductId ) {
 
 	return sitePlan.product_id === planProductId;
 }
+
+export function siteCanUpdateFiles( state, siteId ) {
```

Comment:
Adding jsdoc here for both `siteCanUpdateFiles` and `isMainNetworkSite` would be really useful.